src/libraries/System.Private.Xml/src/System/Xml/Serialization/Xmlcustomformatter.cs

```diff
@@ -405,6 +432,28 @@ internal static DateTime ToTime(string value)
             }
         }
 
+        internal static TimeOnly ToTimeOnly(string value)
+        {
+            if (LocalAppContextSwitches.AllowXsdTimeToTimeOnlyWithOffsetLoss)
+            {
+                return ToTimeOnlyIgnoreOffset(value);
+            }
+
+            // Strictly parse the expected TimeOnly format.
+            return TimeOnly.ParseExact(value, "HH:mm:ss.FFFFFFF", DateTimeFormatInfo.InvariantInfo, DateTimeStyles.AllowLeadingWhite | DateTimeStyles.AllowTrailingWhite);
+        }
+
+        internal static TimeOnly ToTimeOnlyIgnoreOffset(string value)
+        {
+            // Previous workarounds for lack of TimeOnly support included serializing a TimeOnly with 'DataType="time"'.
+            // xsd:time potentially contains an offset designation though, which TimeOnly does not. This would be considered
+            // a loss of data. If the intent was never to include that data, this switch allows for TimeOnly to receive
+            // data from an xsd:time, even if it contains offset information.
+            // Use DateTimeOffset so the time of day is not adjusted for the offset.
+            var dto = DateTimeOffset.ParseExact(value, s_allTimeFormats, DateTimeFormatInfo.InvariantInfo, DateTimeStyles.AllowLeadingWhite | DateTimeStyles.AllowTrailingWhite);
+            return TimeOnly.FromTimeSpan(dto.TimeOfDay);
```

Comment:
This technically allows an even broader range of inputs, it would allow a full date, time, and offset value to be in the xml and we'd only extract the time. As this requires an app context switch to enable, I think this is okay, just wanted to call it out to make sure we agree it's okay.

Reply:
I misread `s_allTimeFormats`  as `s_allDateTimeFormats`